app/src/main/java/com/cpstudy/gogumaproject/MainActivity.kt

```diff
@@ -57,113 +57,23 @@ class MainActivity : AppCompatActivity(), OnMapReadyCallback {
         binding.lifecycleOwner = this
         setContentView(binding.root)
 
-        LocationServices.getFusedLocationProviderClient(this)
-        checkPermission()
-
         val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
         mapFragment.getMapAsync(this)
 
         binding.btnSuggest.setOnClickListener {
-            myLocation()
+            LocationHelper(this).location {
+                mMap.moveCamera(CameraUpdateFactory.newLatLng(LatLng(it.latitude, it.longitude)))
+                mMap.animateCamera(CameraUpdateFactory.zoomTo(15f))
+            }
```

Comment:
최종적으로 여기는 예를들면 이런 형태가 어떨까?
```suggestion
            requestDeviceLocation(activity, onSuccess = {
                 // moveCamera
            })
```

Reply:
어떻게 이렇게 되는건가?

Reply:
위 리뷰들을 반영해서 이런 형태의 메소드면 좋을거 같다는 뜻이야